### PR TITLE
Try to fix #2310 -- Can't create run directory without suid-root

### LINKED ISSUE #2310: Can't create run directory without suid-root
I compiled firejail with `--disable-suid` and tried running it, but it fails to start:
```
Error mkdir: util.c:1014 create_empty_dir_as_root: Permission denied
```
It tries to create its runtime hierarchy in `/run/firejail`, which is not possible as unprivileged user.
If this is not supported, it should check the uid and print a more friendly error message if the user is not root.
If an unprivileged user should be able to run it (without suid-root), maybe it can create required directories in the user's home directory.

### PR DESCRIPTION
**Pull request by rusty-snake**
(untested)

### CODE REVIEW

File: src/firejail/main.c [hunk 1]
```diff
@@ -982,6 +982,14 @@ int main(int argc, char **argv, char **envp) {
 	int arg_caps_cmdline = 0; 	// caps requested on command line (used to break out of --chroot)
 	char **ptr;
 
+#ifndef HAVE_SUID
+	if (geteuid() != 0) {
+		fprintf(stderr, "Error: Firejail need to be SUID.");
```
Comment (glitsj16) [inline src/firejail/main.c:987]: nitpick/typo: Error: Firejail needs to be SUID.

**Review by glitsj16 — APPROVED** [review]
LGTM

File: src/firejail/main.c (hunk 1 above)
Comment (rusty-snake) [reply src/firejail/main.c:987]: fixed, thanks.

File: src/firejail/main.c [hunk 2]
```diff
@@ -982,6 +982,14 @@ int main(int argc, char **argv, char **envp) {
...
+		fprintf(stderr, "Assuming firejail is istalled in /usr/bin, execute the following command as root:");
```
Comment (reinerh) [inline src/firejail/main.c:988]: typo: installed

Comment (rusty-snake) [reply src/firejail/main.c]: fixed, thx.

File: src/firejail/main.c [hunk 3]
```diff
@@ -982,6 +982,14 @@ int main(int argc, char **argv, char **envp) {
 	int arg_caps_cmdline = 0; 	// caps requested on command line (used to break out of --chroot)
 	char **ptr;
 
+#ifndef HAVE_SUID
+	if (geteuid() != 0) {
+		fprintf(stderr, "Error: Firejail needs to be SUID.");
```
Comment (reinerh) [inline src/firejail/main.c:987]: You also want to add newlines (`\n`), as otherwise the error message looks a bit weird (everything in one line).

Comment (rusty-snake) [reply src/firejail/main.c:987]: Absolutely right.

**Review by reinerh — APPROVED** [review]

**Comment by netblue30** [comment]
all in, thanks!

**Pull request merged**